src/main/java/db/parsers/Parser.java

```diff
@@ -10,50 +10,61 @@
 import java.nio.ByteBuffer;
 
 import com.dant.utils.Log;
+import com.dant.utils.Timer;
 
 import db.structure.Column;
 import db.structure.Table;
 
 public abstract class Parser {
 	protected Table schema;
 	protected int lineByteSize; // number of bytes used to store information
-	protected int entryNumber = 0;
+	protected int totalEntryCount = 0;
 	
 	public Parser(Table schema) {
 		this.schema = schema;
 		this.lineByteSize = schema.getLineSize();
 	}
 	
-	public final void parse(InputStream input) {
-		parse(input, -1);
+	public final void parse(InputStream input, boolean appendAtTheEndOfSave) {
+		parse(input, -1, appendAtTheEndOfSave);
 	}
 	
+	protected int showInfoEveryParsedLines = 100_000; // mettre -1 pour désactiver l'affichage
+	
 	/**
 	 * Parse an input stream into an output stream according to a schema with a
 	 * limit of lines (-1 : no limit)
 	 * 
 	 * @param input
 	 * @param limit
 	 */
-	public final void parse(InputStream input, int limit) {
+	public final void parse(InputStream input, int limit, boolean appendAtTheEndOfSave) {
+		int localReadEntryNb = 0;
 		try (
 				BufferedReader bRead = new BufferedReader(new InputStreamReader(input));
-				DataOutputStream bWrite = new DataOutputStream(new BufferedOutputStream(schema.tableToOutputStream()));
+				
+				DataOutputStream bWrite = new DataOutputStream(new BufferedOutputStream(schema.tableToOutputStream(appendAtTheEndOfSave)));
 		) {
-			while (entryNumber != limit) {
+			Timer timeTookTimer = new Timer("Temps écoulé");
+			while (totalEntryCount != limit) {
+				if (localReadEntryNb % showInfoEveryParsedLines == 0 && showInfoEveryParsedLines != -1) {
+					Log.info("Parser : nombre de résultats (local) parsés = " + localReadEntryNb + "   temps écoulé = " + timeTookTimer.getseconds() + " s");
```

Comment:
Tu aurais pu utiliser la fonction timer.pretty() pour ça
```suggestion
					Log.info("Parser : nombre de résultats (local) parsés = " + localReadEntryNb + "   temps écoulé = " + timeTookTimer.pretty());
```

Reply:
Approuvé ! :3

Reply:
Parler de la manière dont fonctionne l'IndextTreeDic par téléphone me semble être une bonne idée, perso, en tout cas ! :=